core/src/main/java/com/sequenceiq/distrox/v1/distrox/service/upgrade/DistroXUpgradeAvailabilityService.java

```diff
@@ -69,6 +69,10 @@ public UpgradeV4Response checkForUpgrade(NameOrCrn nameOrCrn, Long workspaceId,
         UpgradeV4Response response = stackOperations.checkForClusterUpgrade(stack, workspaceId, request);
         List<ImageInfoV4Response> filteredCandidates = filterCandidates(stack, request, response.getUpgradeCandidates());
         response.setUpgradeCandidates(filteredCandidates);
+        if (filteredCandidates.isEmpty()) {
```

Comment:
UpgradeV4Response response could be passed instead of response.getUpgradeCandidates() so that 
- we could set the reason during filtering in method filterForDatalakeVersion() --> message will always be 100% accurate
- we could add the versions of the DH and the DL to provide more info --> message will be more useful